lib/rex/post/meterpreter/extensions/stdapi/sys/registry.rb

```diff
@@ -195,10 +212,55 @@ def Registry.set_value(hkey, name, type, data)
     return true
   end
 
+  def Registry.set_value_direct(root_key, base_key, name, type, data, perm = KEY_WRITE)
+    request = Packet.create_request('stdapi_registry_set_value_direct')
+
+    request.add_tlv(TLV_TYPE_ROOT_KEY, root_key)
+    request.add_tlv(TLV_TYPE_BASE_KEY, base_key)
+    request.add_tlv(TLV_TYPE_PERMISSION, perm)
+    request.add_tlv(TLV_TYPE_VALUE_NAME, name)
+    request.add_tlv(TLV_TYPE_VALUE_TYPE, type)
+
+    if (type == REG_SZ)
+      data += "\x00"
+    elsif (type == REG_DWORD)
+      data = [ data.to_i ].pack("V")
```

Comment:
Ideally not spaces after/before `[` and `]` https://github.com/bbatsov/ruby-style-guide#no-spaces-braces